src/zenml/integrations/label_studio/label_studio_utils.py

```diff
@@ -36,22 +43,38 @@ def convert_pred_filenames_to_task_ids(
         List of predictions using task ids as reference.
     """
     filename_id_mapping = {
-        os.path.basename(
-            urlparse(task["data"][filename_reference]).path
-        ): task["id"]
+        clean_url(task["data"][filename_reference]): task["id"]
         for task in tasks
     }
+
     # GCS and S3 URL encodes filenames containing spaces, requiring this
     # separate encoding step
-    if storage_type in {"gcs", "s3"}:
+    if storage_type == "gcs":
         preds = [
-            {"filename": quote(pred["filename"]), "result": pred["result"]}
+            {
+                "filename": quote(pred["filename"]).split("//")[1],
+                "result": pred["result"],
+            }
             for pred in preds
         ]
+    elif storage_type == "s3":
+        preds = [
+            {
+                "filename": "/".join(
+                    quote(pred["filename"]).split("//")[1].split("/")[1:]
+                ),
+                "result": pred["result"],
+            }
+            for pred in preds
+        ]
+
     return [
         {
             "task": int(
-                filename_id_mapping[os.path.basename(pred["filename"])]
+                filename_id_mapping[
+                    urlparse(pred["filename"]).netloc
+                    + urlparse(pred["filename"]).path
```

Comment:
would be cleaner if you define `parsed_url = urlparse(pred["filename"])` in a separate line

Reply:
Would you prefer that (in a standard for loop) over a list comprehension? To my eyes somehow the list comprehension implementation is nicer than:

```python
output = []
for pred in preds:
    parsed_url = urlparse(pred["filename"])
    task = int(filename_id_mapping[parsed_url.netloc + parsed_url.path])
    result = pred["result"]
    output.append({"task": task, "result": result})

return output
```

Reply:
ok up to you :)